source/extensions/filters/common/ratelimit/ratelimit_registration.cc

```diff
@@ -0,0 +1,34 @@
+#include "extensions/filters/common/ratelimit/ratelimit_registration.h"
+
+namespace Envoy {
+namespace Extensions {
+namespace Filters {
+namespace Common {
+namespace RateLimit {
+
+// Singleton registration via macro defined in envoy/singleton/manager.h
```

Comment:
Ultimately, we just want the client in the singleton not the config. Config was added in the singleton as an interim step, which will be removed once we remove rate limit config from bootstrap. Then this whole file will go away. So it will be a least disruptive change. The validation will be added in the filter config code. Will documentation around that.

Reply:
Sorry I don't follow. Why can't you store the client in the singleton manager in this change? It should be fine and what you want in the future anyway.

Reply:
I am storing client in singleton manager already in this change. I am just saying i am also storing config in the interim.